techbuzz/src/main/java/com/sivalabs/techbuzz/users/domain/services/UserService.java

```diff
@@ -66,4 +77,22 @@ public void verifyEmail(String email, String token) {
         user.setVerificationToken(null);
         userRepository.updateVerificationStatus(user);
     }
+
+    public Optional<UserDTO> getUserDTO(String emailID) {
+        return this.getUserByEmail(emailID).map(userDTOMapper::toDTO);
+    }
+
+    public void sendVerificationEmail(HttpServletRequest request, UserDTO userDTO) {
```

Comment:
Any HTTP related things should not leak into "domain" package. We should extract the necessary information from HttpServletRequest in Controller and then send the data to Service.

Reply:
Got it . Thanks updated.